### PR TITLE
added sql_live_mount function

### PR DESCRIPTION
**Pull request by promiseowolabi**
# Description

Created a new function to live mount SQL databases

## Related Issue

https://github.com/rubrikinc/rubrik-sdk-for-python/issues/121

## Motivation and Context

This is an enhancement to be able to do live mount for test, dev and recovery workflows

## How Has This Been Tested?

Only tested locally on demo clusters i.e. recovery point validation

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

Go over all the following points, and put an `x` in all the boxes that apply. If you're unsure about any of these, don't hesitate to ask. We're here to help!
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **[CONTRIBUTION](https://github.com/rubrikinc/rubrik-sdk-for-python/blob/master/CONTRIBUTING.md)** document.
- [ ] I have updated the CHANGELOG file accordingly for the version that this merge modifies.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: rubrik_cdm/data_management.py [hunk 1]
```diff
@@ -885,3 +886,81 @@ def get_sla_objects(self, sla, object_type, timeout=15):
                         sla, object_type))
 
             return vm_name_id
+
+    def time_in_range(self, start, end, x):
```
Comment (drew-russell) [inline rubrik_cdm/data_management.py]: `time_in_range` updates:

* function should be internal (i.e named `_time_in_range`)
* let's use a more descriptive variable name than `x`
* need full docstring similar to the rest of the functions

File: rubrik_cdm/data_management.py [hunk 2]
```diff
@@ -885,3 +886,81 @@ def get_sla_objects(self, sla, object_type, timeout=15):
...
+
+        Keyword Arguments:
+            timeout {int} -- The number of seconds to wait to establish a connection the Rubrik cluster before returning a timeout error. (default: {15})
+
+        Returns:
+            dict -- The full response of `POST /v1/mssql/db/{id}/mount`.
+        """
+        
+        mssql_host_id = self.object_id(sql_host, 'physical_host', timeout=timeout)
+
```
Comment (drew-russell) [inline rubrik_cdm/data_management.py]: Add a logging message for 

`mssql_instance = self.get('v1', '/mssql/instance?primary_cluster_id=local&root_id={}'.format(mssql_host_id), timeout)`

that describes what it is doing

File: rubrik_cdm/data_management.py [hunk 3]
```diff
@@ -885,3 +886,81 @@ def get_sla_objects(self, sla, object_type, timeout=15):
...
+        mssql_instance = self.get(
+            'v1', '/mssql/instance?primary_cluster_id=local&root_id={}'.format(mssql_host_id), timeout)
+
+        for instance in mssql_instance['data']:
+            if instance['name'] == sql_instance:
+                sql_instance_id = instance['id']
+
+        mssql_db = self.get('v1', '/mssql/db?primary_cluster_id=local&instance_id={}'.format(sql_instance_id), timeout)
```
Comment (drew-russell) [inline rubrik_cdm/data_management.py]: Add a logging message for 

`mssql_db = self.get('v1', '/mssql/db?primary_cluster_id=local&instance_id={}'.format(sql_instance_id), timeout)`


that describes what it is doing

File: rubrik_cdm/data_management.py [hunk 4]
```diff
@@ -885,3 +886,81 @@ def get_sla_objects(self, sla, object_type, timeout=15):
...
+        """
+        
+        mssql_host_id = self.object_id(sql_host, 'physical_host', timeout=timeout)
+
+        mssql_instance = self.get(
+            'v1', '/mssql/instance?primary_cluster_id=local&root_id={}'.format(mssql_host_id), timeout)
+
+        for instance in mssql_instance['data']:
+            if instance['name'] == sql_instance:
+                sql_instance_id = instance['id']
```
Comment (drew-russell) [inline rubrik_cdm/data_management.py]:
```
for instance in mssql_instance['data']:
            if instance['name'] == sql_instance:
                sql_instance_id = instance['id']
```

Is there a possibility of not finding a match here?

File: rubrik_cdm/data_management.py [hunk 5]
```diff
@@ -885,3 +886,81 @@ def get_sla_objects(self, sla, object_type, timeout=15):
...
+
+        mssql_db = self.get('v1', '/mssql/db?primary_cluster_id=local&instance_id={}'.format(sql_instance_id), timeout)
+
+        for db in mssql_db['data']:
```
Comment (drew-russell) [inline rubrik_cdm/data_management.py]:
```
 for db in mssql_db['data']:
            if db['name'] == sql_db:
                mssql_id = db['id']
```

Similar to above....is there a possibility of not returning a match here?

Comment (promiseowolabi) [reply rubrik_cdm/data_management.py]: The match is restricted to the databases on the sql_instance input, there should always be a unique match within the instance. I will add a invalid parameter exceptions for when the database doesn't exist

File: rubrik_cdm/data_management.py (hunk 4 above)
Comment (promiseowolabi) [reply rubrik_cdm/data_management.py]: To get the right database we need the host_id (from hostname) and instance name (its possible to have the same database name on different instances on the same host). Is there a possibility of not finding a match here? I don't think there is, we are working with the system assigned ids.
I will add a invalid parameter exceptions for when the instance doesn't exist

Comment (promiseowolabi) [reply rubrik_cdm/data_management.py]: Also, instance names are unique on the same host.

File: rubrik_cdm/data_management.py [hunk 6]
```diff
@@ -885,3 +885,108 @@ def get_sla_objects(self, sla, object_type, timeout=15):
...
+            sql_instance {str} -- The SQL instance name with the database you wish to Live Mount.
+            sql_host {str} -- The SQL Host of the database/instance to Live Mount.
+            mount_name {str} -- The name given to the Live Mounted database i.e. AdventureWorks_Clone.
+            timeout {int} -- The number of seconds to wait to establish a connection the Rubrik cluster before returning a timeout error. (default: {30})
+
+        Returns:
+            dict -- The full response of `POST /v1/mssql/db/{id}/mount`.
+        """
+        
+        from datetime import datetime
```
Comment (drew-russell) [inline rubrik_cdm/data_management.py]: Move `from datetime import datetime` to the top of the file

Comment (promiseowolabi) [reply rubrik_cdm/data_management.py]: I have now

Comment (promiseowolabi) [reply rubrik_cdm/data_management.py]: And updated the vmware and sql sample, copy and paste mistake.

**Pull request merged**